packages/apps/reputation-oracle/server/src/modules/cron-job/cron-job.service.ts

```diff
@@ -187,61 +124,7 @@ export class CronJobService {
     );
 
     try {
-      const escrowCompletionTrackingEntities =
-        await this.escrowCompletionTrackingRepository.findByStatus(
-          EscrowCompletionTrackingStatus.PENDING,
-        );
-
-      for (const escrowCompletionTrackingEntity of escrowCompletionTrackingEntities) {
-        try {
-          const signer = this.web3Service.getSigner(
-            escrowCompletionTrackingEntity.chainId,
-          );
-          const escrowClient = await EscrowClient.build(signer);
-
-          const escrowStatus = await escrowClient.getStatus(
-            escrowCompletionTrackingEntity.escrowAddress,
-          );
-          if (escrowStatus === EscrowStatus.Pending) {
-            if (!escrowCompletionTrackingEntity.finalResultsUrl) {
-              const { url, hash } = await this.payoutService.saveResults(
-                escrowCompletionTrackingEntity.chainId,
-                escrowCompletionTrackingEntity.escrowAddress,
-              );
-
-              escrowCompletionTrackingEntity.finalResultsUrl = url;
-              escrowCompletionTrackingEntity.finalResultsHash = hash;
-              await this.escrowCompletionTrackingRepository.updateOne(
-                escrowCompletionTrackingEntity,
-              );
-            }
-
-            await this.payoutService.executePayouts(
-              escrowCompletionTrackingEntity.chainId,
-              escrowCompletionTrackingEntity.escrowAddress,
-              escrowCompletionTrackingEntity.finalResultsUrl,
-              escrowCompletionTrackingEntity.finalResultsHash,
-            );
-          }
-
-          escrowCompletionTrackingEntity.status =
-            EscrowCompletionTrackingStatus.PAID;
-          await this.escrowCompletionTrackingRepository.updateOne(
-            escrowCompletionTrackingEntity,
-          );
-        } catch (err) {
-          const errorId = uuidv4();
-          const failureDetail = `${ErrorWebhook.PendingProcessingFailed} (Error ID: ${errorId})`;
-          this.logger.error(
-            `Error processing escrow completion tracking. Error ID: ${errorId}, Escrow completion tracking ID: ${escrowCompletionTrackingEntity.id}, Reason: ${failureDetail}, Message: ${err.message}`,
-          );
-          await this.escrowCompletionTrackingService.handleEscrowCompletionTrackingError(
-            escrowCompletionTrackingEntity,
-            failureDetail,
-          );
-          continue;
-        }
-      }
+      await this.webhookService.processPendingEscrowCompletion();
```

Comment:
Should `processPendingEscrowCompletion` be part of `EscrowCompletiontTracking` service?

Reply:
Yeah, I thought about that, I ran into a cyclic dependency issue where the Escrow module uses the Webhook module, and the Webhook module uses the Escrow module. Do you have any ideas on how to refactor the process to avoid cyclic dependencies?